### PR TITLE
Issue #1657: Reduce Window copying

### PR DESCRIPTION
**Pull request by hawkfish**
The R/last example produces 400K 10-element blocks,
which we were copying more than once. This change
moves hash partitioning into the sorting code
so that larger blocks can be built up efficiently in place.
The performance gain on the example is about 2x.
(1.96s => 1.05s with 8 threads)

### CODE REVIEW

File: src/execution/operator/aggregate/physical_window.cpp [hunk 1]
```diff
@@ -459,34 +472,85 @@ static void SortCollectionForPartition(PhysicalWindowOperatorState &state, Bound
...
+			idx_t bin_size = 0;
+			for (idx_t i = 0; i < hash_size; ++i) {
+				if ((hash_data[i] & hash_mask) == hash_bin) {
+					sel.set_index(bin_size++, i);
+				}
+			}
+
+			// Flush the partition chunks if we would overflow
+			if (over_partition.size() + bin_size > STANDARD_VECTOR_SIZE) {
+				local_sort_state.SinkChunk(over_partition, payload_partition);
```
Comment (Mytherin) [inline src/execution/operator/aggregate/physical_window.cpp:494]: This appears to be uncovered, could you try to add a test case that triggers this?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks great. One minor comment, otherwise I think it is good to go!

File: src/execution/operator/aggregate/physical_window.cpp (hunk 1 above)
Comment (hawkfish) [reply src/execution/operator/aggregate/physical_window.cpp:494]: Amusingly this is the piece of code I was profiling for 4 days!  But yes, it takes a test case with multiple partitions that are over 2kb. I'll throw something together.

**Comment by Mytherin** [comment]
Thanks! Looks great.

**Pull request merged**